tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -1233,3 +1284,218 @@ def apply_bspline_transform(
 
     sitk_registered_image_sitk = resampler.Execute(moving_image_sitk)
     return sitk.GetArrayFromImage(sitk_registered_image_sitk)
+
+
+class AffineWSITransformer:
+    """Resampling regions from a whole slide image.
+
+    This class is used to resample tiles/patches from a whole slide image
+    using transformation.
+
+    Example:
+        >>> from tiatoolbox.tools.registration.wsi_registration import (
+        >>> AffineWSITransformer
+        >>> )
+        >>> from tiatoolbox.wsicore.wsireader import WSIReader
+        >>> wsi_reader = WSIReader.open(input_img=sample_ome_tiff)
+        >>> transform_level0 = np.eye(3)
+        >>> tfm = AffineWSITransformer(wsi_reader, transform_level0)
+        >>> output = tfm.read_rect(location, size, resolution=resolution, units="level")
+
+    """
+
+    def __init__(self, wsi_reader: WSIReader, transform: np.ndarray) -> None:
```

Comment:
```suggestion
    def __init__(self, img: WSIReader, transform: np.ndarray) -> None:
```

Reply:
I think 'reader' might be better. Calling it 'img' could be a little misleading.

Reply:
We are using img in model engines.

Reply:
@John-P  In the revised jupyter notebook you are using `reader`, so I am OK as far as we are consistent.

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/faa20500c80e846f1ae5d8e62a8a36a8c832dbc6/tiatoolbox/tools/registration/wsi_registration.py#L1313-L1323